internal/reachability/client.go

```
@@ -0,0 +1,196 @@
+			Attributes: StartReachabilityAttributes{
+				BundleID: bundleID,
+			},
+		},
+	}
+
+	buff := bytes.NewBuffer(nil)
+	if err := json.NewEncoder(buff).Encode(body); err != nil {
+		return uuid.Nil, fmt.Errorf("failed to encode reachability request body: %w", err)
+	}
+
+	url := fmt.Sprintf("%s/hidden/orgs/%s/reachability?version=%s", sec.cfg.BaseURL, orgID, apiVersion)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, buff)
+	if err != nil {
+		return uuid.Nil, fmt.Errorf("failed to create reachability request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/vnd.api+json")
+
+	res, err := sec.httpClient.Do(req)
+	if err != nil {
+		return uuid.Nil, fmt.Errorf("error making reachability request: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusCreated {
+		return uuid.Nil, handleUnexpectedStatusCodes(res.Body, res.StatusCode, res.Status, "reachability analysis")
+	}
+
+	var respBody StartReachabilityResponseBody
+	if err := json.NewDecoder(res.Body).Decode(&respBody); err != nil {
+		return uuid.Nil, fmt.Errorf("failed to decode reachability response body: %w", err)
+	}
+
+	return respBody.Data.ID, nil
+}
+
+// WaitForReachabilityAnalysis polls for the completion of a reachability analysis until it finishes, fails, or times out.
+// It returns nil when the analysis completes successfully, or an appropriate error for other outcomes.
+func (sec *SCAEngineClient) WaitForReachabilityAnalysis(ctx context.Context, orgID OrgID, reachabilityID ID) error {
+	timeout := time.After(sec.cfg.PollTimeout)
+	t := time.NewTicker(sec.cfg.PollInterval)
+	defer t.Stop()
+
+	for {
+		status, err := sec.getReachabilityAnalysisStatus(ctx, orgID, reachabilityID)
+		if err != nil {
+			return err
+		}
+		switch status {
+		case ScanStatusDone:
+			return nil
+		case ScanStatusFailed:
+			return ErrScanFailed
+		case ScanStatusTimeout:
+			return ErrScanTimedOut
+		case ScanStatusInProgress:
+		case ScanStatusUnknown:
```

I think we can fail on unknown, this shouldn't happen